support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -181,52 +277,71 @@ function userInBreakpoint(audience: Audience): boolean {
 	return false;
 }
 
-function userInTest(
+type NoParticipation = { type: 'NO_PARTICIPATION' };
+
+type ActiveParticipation =
+	| { type: 'ACTIVE_PARTICIPATION' }
+	| {
+			type: 'REFERRER_CONTROLLED_ACTIVE_PARTICIPATION';
+			acquisitionDataTest: AcquisitionABTest;
+	  };
+
+type UserParticipation = NoParticipation | ActiveParticipation;
+
+function getUserParticipation(
 	test: Test,
 	testId: string,
 	mvtId: number,
 	country: IsoCountry,
 	countryGroupId: CountryGroupId,
 	acquisitionDataTests: AcquisitionABTest[] | undefined,
-): boolean {
+): UserParticipation {
 	const { audiences, referrerControlled } = test;
 
 	if (cookie.get('_post_deploy_user')) {
-		return false;
+		return NO_PARTICIPATION;
 	}
 
 	const audience =
 		audiences[country] ?? audiences[countryGroupId] ?? audiences.ALL;
 
 	if (!audience) {
-		return false;
+		return NO_PARTICIPATION;
 	}
 
 	if (referrerControlled) {
-		const isSfdV2Test =
-			testId === 'SFD_V2' &&
-			!!acquisitionDataTests &&
-			acquisitionDataTests.some((acquisitionDataTest) =>
-				acquisitionDataTest.name.startsWith('SFD_V2'),
-			);
+		// For referrer controlled tests we have to search through the tests in the acquisition
+		// data to find a match. We use the `startsWith` method to support test campaigns all
+		// with a common prefix.
+		const acquisitionDataTest = acquisitionDataTests?.find(
+			(acquisitionDataTest) => acquisitionDataTest.name.startsWith(testId),
+		);
 
-		if (isSfdV2Test) {
-			return isSfdV2Test;
+		if (!acquisitionDataTest) {
+			return NO_PARTICIPATION;
 		}
 
-		return (
-			!!acquisitionDataTests &&
-			acquisitionDataTests.some(
-				(acquisitionDataTest) => acquisitionDataTest.name === testId,
-			)
-		);
+		return referrerControlledActiveParticipation(acquisitionDataTest);
```

Comment:
I thought we could remove the special case logic for the single front door test and just make it a feature of `referrerControlled` tests to support test campaigns

Reply:
I've added a new test case for this behaviour